lxd/device/nic_routed.go

```diff
@@ -497,3 +502,36 @@ func (d *nicRouted) ipv6HostAddress() string {
 
 	return nicRoutedIPv6GW
 }
+
+func (d *nicRouted) isUniqueWithGatewayAutoMode(instConf instance.ConfigReader) error {
+
+	if instConf.ExpandedDevices() != nil && d.hasAutoGatewayConfiguration(d.config) {
+		routedNicDeviceNames := []string{d.config["name"]}
+
+		nicDevices := instConf.ExpandedDevices().FilterByConfigProperties(map[string]string{"nictype": "routed"})
```

Comment:
@my5t3ry I think I'd be tempted to handle all the validation in this function and avoid the need for `FilterByConfigProperties` function, and instead keeping all the logic in one function and running the validation for each gateway key in a loop.

e.g.

```go
	instDevs := instConf.ExpandedDevices()
	for k, v := range []string{"ipv4.gateway", "ipv6.gateway"} {
		ourGW := d.config[k]

		if ourGW != "auto" {
			continue // nothing to do as auto not being used.
		}

		// Check other routed NIC devices don't have auto set.
		for nicName, nicConfig := range instDevs {
			if nicName == d.name {
				continue // Skip ourselves.
			}

			if nicConfig[k] == "auto" {
				return fmt.Errorf("Existing NIC %q already uses %q in auto mode", nicName, k)
			}
		}
	}
```